### PR TITLE
feat: Feat/add google sheets metdata

### LINKED ISSUE #21759: [Feature]: Limiting Gsheet - List selected files as well as account name on datasource review page (Fetch Selected Files)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

On datasource review page, I should be able to see:
- Account info which is used for google sheet authorisation

For specific sheets modality, I should be able to see:
- Names of selected sheets

### Why should this be worked on?

When I look at datasource review page, I should be able to see the list of files selected, so that I can know which files have authorisation

### PR DESCRIPTION
**Pull request by Irongade**
This PR does three things:

1. Cleans up the Datasource Form page and makes the code more readable and discoverable
2. Adds the Account information for google sheets
3. Adds a way to render form configs specifically in view or edit modes.

> Add a TL;DR when description is extra long (helps content team)

Fixes #21759 

- New feature (non-breaking change which adds functionality)

- Manual
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/pages/Editor/DataSourceEditor/DBForm.tsx [hunk 1]
```diff
@@ -203,7 +212,20 @@ class DatasourceDBEditor extends JSONtoForm<Props> {
                 {""}
               </>
             )}
-            {viewMode && <Connected />}
+            {viewMode && (
+              <ViewModeWrapper>
+                <Connected />
+                <div style={{ marginTop: "30px" }}>
```
Comment (sneha122) [inline app/client/src/pages/Editor/DataSourceEditor/DBForm.tsx:218]: Is this marginTop needed?

File: app/client/src/pages/Editor/DataSourceEditor/DatasourceSection.tsx [hunk 2]
```diff
@@ -123,6 +125,15 @@ export default class RenderDatasourceInformation extends React.Component<{
                 }
               }
 
+              if (
+                !value &&
+                !!viewMode &&
+                "comparison" in section.hidden &&
+                section.hidden.comparison === "VIEW_MODE"
```
Comment (sneha122) [inline app/client/src/pages/Editor/DataSourceEditor/DatasourceSection.tsx:132]: Can we use enum here instead for "VIEW_MODE"?

File: app/client/src/pages/Editor/DataSourceEditor/DBForm.tsx (hunk 1 above)
Comment (Irongade) [reply app/client/src/pages/Editor/DataSourceEditor/DBForm.tsx:218]: Yes it does,

**Review by sumitsum — CHANGES_REQUESTED** [review]

File: app/server/appsmith-plugins/googleSheetsPlugin/src/main/resources/form.json [hunk 3]
```diff
@@ -72,6 +72,17 @@
...
+        {
+          "label": "Account",
+          "configProperty": "datasourceConfiguration.properties[0].key",
+          "controlType": "INPUT_TEXT",
+          "isRequired": false,
+          "hidden": {
+            "comparison": "VIEW_MODE",
+            "value": false
+          },
+          "initialValue": "Authorize datasource to fetch account name"
```
[the remark this answers is not in the document]
Comment (sneha122) [reply app/server/appsmith-plugins/googleSheetsPlugin/src/main/resources/form.json:84]: Screenshot 2023-04-25 at 5 21 30 PM


@sumitsum It is as per designs. CC: @Irongade

File: app/server/appsmith-plugins/googleSheetsPlugin/src/main/resources/form.json [hunk 4]
```diff
@@ -72,6 +72,17 @@
           "hidden": true,
           "initialValue": "authorization_code"
         },
+        {
+          "label": "Account",
+          "configProperty": "datasourceConfiguration.properties[0].key",
```
[the remark this answers is not in the document]
Comment (Irongade) [reply app/server/appsmith-plugins/googleSheetsPlugin/src/main/resources/form.json]: Done

File: app/server/appsmith-plugins/googleSheetsPlugin/src/main/resources/form.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (sumitsum) [reply app/server/appsmith-plugins/googleSheetsPlugin/src/main/resources/form.json:84]: @sneha122 its ok, I am pretty sure we are making it better by prefix with `Please` :)

**Review by sumitsum — APPROVED** [review]

**Comment by sumitsum** [comment]
server side changes LGTM.

**Review by sneha122 — APPROVED** [review]

**Review by sneha122 — APPROVED** [review]

**Review by sumitsum — APPROVED** [review]

**Review by sneha122 — APPROVED** [review]

**Review by sneha122 — APPROVED** [review]

**Review by sumitsum — APPROVED** [review]

**Pull request merged**